lte/gateway/c/oai/tasks/mme_app/mme_app_state_converter.cpp

```diff
@@ -134,24 +151,53 @@ void MmeNasStateConverter::guti_table_to_proto(
     } else {
       OAILOG_ERROR(LOG_MME_APP, "Key %s not in guti_ue_context_htbl", guti_str);
     }
+    OAILOG_DEBUG(
+        LOG_MME_APP, "guti_str:%s mme_ue_id:%d\n", guti_str.c_str(), mme_ue_id);
   }
   FREE_OBJ_HASHTABLE_KEY_ARRAY(key_array_p);
 }
 
+void MmeNasStateConverter::mme_app_convert_string_to_guti(
+    guti_t* guti_p, const std::string& guti_str) {
+  int idx         = 0;
+  std::size_t num = 1;
+#define HEX_BASE_VAL 16
+    guti_p->gummei.plmn.mcc_digit1 =
+        std::stoul(guti_str.substr(idx++, num), &num, HEX_BASE_VAL);
+    guti_p->gummei.plmn.mcc_digit2 =
+        std::stoul(guti_str.substr(idx++, num), &num, HEX_BASE_VAL);
+    guti_p->gummei.plmn.mcc_digit3 =
+        std::stoul(guti_str.substr(idx++, num), &num, HEX_BASE_VAL);
+    guti_p->gummei.plmn.mnc_digit1 =
+        std::stoul(guti_str.substr(idx++, num), &num, HEX_BASE_VAL);
+    guti_p->gummei.plmn.mnc_digit2 =
+        std::stoul(guti_str.substr(idx++, num), &num, HEX_BASE_VAL);
+    guti_p->gummei.plmn.mnc_digit3 =
+        std::stoul(guti_str.substr(idx++, num), &num, HEX_BASE_VAL);
+    num = 4;
+    guti_p->gummei.mme_gid =
+        std::stoul(guti_str.substr(idx, num), &num, HEX_BASE_VAL);
+    idx += num;
+    num = 1;
```

Comment:
If we fix the mme_code to be written in 2 characters, then we need to read 2 characters here.

Reply:
MME code is of 1 byte